addons/project/models/project_update.py

```diff
@@ -0,0 +1,253 @@
+        return self._build_description(project_id)
+
+    name = fields.Char("Title", required=True, tracking=True)
+    status_id = fields.Many2one("project.update.status", copy=False, default=_get_default_status, required=True, tracking=True)
+    color = fields.Integer(related="status_id.color")
+    progress = fields.Integer(tracking=True)
+    progress_percentage = fields.Float(compute="_compute_progress_percentage")
+    user_id = fields.Many2one('res.users', string="Author", required=True, default=lambda self: self.env.user)
+    description = fields.Html(default=_get_default_description)
+    date = fields.Date(default=fields.Date.context_today, tracking=True)
+    project_id = fields.Many2one("project.project", required=True, default=_get_default_project_id)
+    name_cropped = fields.Char(compute="_compute_name_cropped")
+
+    def _compute_progress_percentage(self):
+        for u in self:
+            u.progress_percentage = u.progress / 100
+
+    @api.depends('name')
+    def _compute_name_cropped(self):
+        for u in self:
+            u.name_cropped = len(u.name) > 60 and u.name[:57] + '...' or u.name
+
+    # ---------------------------------
+    # Build default description
+    # ---------------------------------
+    @api.model
+    def _build_description(self, project_id):
+        template = self.env.ref('project.project_update_default_description')
+        return template._render(self._get_template_values(project_id=project_id), engine='ir.qweb')
+
+    @api.model
+    def _get_template_values(self, project_id):
+        return {
+            'user': self.env.user,
+            'project': self.env['project.project'].browse(project_id),
+            'tasks': self._get_tasks_values(project_id=project_id),
+            'milestones': self._get_milestone_values(project_id=project_id)
+        }
+
+    @api.model
+    def _get_tasks_values(self, project_id):
+        Task = self.env['project.task']
+        action_url = '%s/web#menu_id=%s&action=%s&active_id=%s' % (
+            self.env['ir.config_parameter'].sudo().get_param('web.base.url'),
+            self.env.ref('project.menu_projects').id,
+            self.env.ref('project.action_project_task_burndown_chart_report').id,
+            project_id
+        )
+        return {
+            'open_tasks': Task.search_count([('display_project_id', '=', project_id), ('stage_id.fold', '=', False), ('stage_id.is_closed', '=', False)]),
+            'total_tasks': Task.search_count([('display_project_id', '=', project_id)]),
+            'created_tasks': Task.search_count([('display_project_id', '=', project_id), ('create_date', '>', fields.Datetime.now() + timedelta(days=-30))]),
+            'closed_tasks': self._get_last_stage_changes(project_id=project_id),
+            'action': {
+                'url_default': action_url,
+                'name': 'action_burndown_open_tasks',
+            }
+        }
+
+    @api.model
+    def _get_last_stage_changes(self, project_id):
```

Comment:
Questions: 

Make a SQL request directly overwrites the ir.rules for instance. Are you sure we won't count "state changes" for records on which a user don't have access to ?

Reply:
Fixed here : https://github.com/odoo-dev/odoo/commit/f424fd2f8e783b15e2ac7b978c67b1c9a0127f91#diff-0872ed6ea72d0e32f902da38133aa49764c7790cb2f534dee6eba215294c6388R131-R133